libbpfgo.go

```diff
@@ -531,6 +531,45 @@ func CreateMap(mapType MapType, mapName string, keySize, valueSize, maxEntries i
 	}, nil
 }
 
+// GetMapByName gets an existing map
+// Can be used for populating BPF array of maps or hash of maps. Currently, the returned map can only be updated.
+func GetMapByName(name string) (*BPFMap, error) {
+	startId := C.uint(0)
+	nextId := C.uint(0)
+
+	for {
+		err := C.bpf_map_get_next_id(startId, &nextId)
+		if err != 0 {
+			return nil, fmt.Errorf("could not get the map: %w", syscall.Errno(-err))
+		}
+
+		startId = nextId + 1
+
+		fd := C.bpf_map_get_fd_by_id(nextId)
+		if fd < 0 {
+			return nil, fmt.Errorf("could not get the file descriptor of %s", name)
+		}
+
+		info := C.struct_bpf_map_info{}
+		infolen := C.uint(unsafe.Sizeof(info))
+		err = C.bpf_obj_get_info_by_fd(fd, unsafe.Pointer(&info), &infolen)
+		if err != 0 {
+			return nil, fmt.Errorf("could not get the map info: %w", syscall.Errno(-err))
+		}
+
+		mapName := C.GoString((*C.char)(unsafe.Pointer(&info.name[0])))
+		if mapName != name {
+			continue
+		}
```

Comment:
And to make sure we're dealing with the right BPF map, I assume we must check the pin path, instead of its name. There might be different maps with the same name (from different BPF objects/origins).

![image](https://github.com/aquasecurity/libbpfgo/assets/3372117/bbaaa1ba-805d-4959-a141-64624fb1c440)

Reply:
Yes, it is a problem 🤔 How can I get the bin path of the map? We can also return an error like `MultipleMapFoundsError` or a list of maps.

Reply:
> How can I get the bin path of the map?

Libbpf holds that information, but we're not treating with libbpf generated maps.

The pinning sets `attr.pathname`. But further, retrieving map information, unfortunately bpf_map_info has no `filepath` field.

![image](https://github.com/aquasecurity/libbpfgo/assets/3372117/aade6f35-a1f0-4d6a-8990-e6f4f56fd67b)

There's another issue with the pin path approach: maps from other origin may not have been pinned.

> We can also return an error like `MultipleMapFoundsError` or a list of maps.

Maybe return a slice of map IDs, so the caller would decide which to choose based on later access to the respective information (type, key/value size, flags, etc.).

Reply:
> It looks good 👍 I can update the PR to return a list of Maps rather the first map matched the name criteria.

Excellent. I would just ask you to wait a bit. I'm doing local testing and looking at splitting maps managed by libbpf and user-created ones (or from another origin) to align in the API.

Reply:
@aymericDD sorry about the delay.

We have merged major map changes, which provide helpers for your attempt. The logic should be in `map-low.go` as that is where the low-level API for maps resides. As agreed, `GetMapsByName()` should return a slice of map IDs.

Reply:
Nice! I will take a look 👍